Ginger/GingerCore/ALM/Qtest/ImportFromQtest.cs

```diff
@@ -0,0 +1,1128 @@
+            return;
+        }
+
+        public static void UpdateBusinessFlow(ref BusinessFlow busFlow, List<QtestTest> tcsList)
+        {
+            if ((busFlow == null) || (tcsList == null) || (tcsList.Count < 1)) return;
+            Dictionary<string, string> busVariables = new Dictionary<string, string>();
+            int startGroupActsIndxInBf = 0;
+
+            busFlow.Activities.Clear();
+            busFlow.ActivitiesGroups.Clear();
+
+            foreach (QtestTest tc in tcsList)
+            {
+                //check if the TC is already exist in repository
+                ActivitiesGroup tcActivsGroup = new ActivitiesGroup();
+
+                tcActivsGroup = new ActivitiesGroup();
+                tcActivsGroup.Name = tc.TestName;
+                if (tc.LinkedTestID == null || tc.LinkedTestID == string.Empty)
+                {
+                    tcActivsGroup.ExternalID = tc.TestID;
+                    tcActivsGroup.ExternalID2 = tc.TestID;
+                }
+                else
+                {
+                    tcActivsGroup.ExternalID = tc.LinkedTestID;
+                    tcActivsGroup.ExternalID2 = tc.TestID; //original TC ID will be used for uploading the execution details back to QC
+                    tcActivsGroup.Description = tc.Description;
+                }
+                busFlow.AddActivitiesGroup(tcActivsGroup);
+
+                //Add the TC steps as Activities if not already on the Activities group
+                foreach (QtestTestStep step in tc.Steps)
+                {
+                    Activity stepActivity;
+                    bool toAddStepActivity = false;
+
+                    //check if mapped activity exist in repository
+                    Activity repoStepActivity =(Activity) GingerActivitiesRepo.Where(x => x.ExternalID == step.StepID).FirstOrDefault();
+                    if (repoStepActivity != null)
+                    {
+                        //check if it is part of the Activities Group
+                        ActivityIdentifiers groupStepActivityIdent = (ActivityIdentifiers)tcActivsGroup.ActivitiesIdentifiers.Where(x => x.ActivityExternalID == step.StepID).FirstOrDefault();
+                        if (groupStepActivityIdent != null)
+                        {
+                            //already in Activities Group so get link to it
+                            stepActivity =(Activity) busFlow.Activities.Where(x => x.Guid == groupStepActivityIdent.ActivityGuid).FirstOrDefault();
+                        }
+                        else//not in ActivitiesGroup so get instance from repo
+                        {
+                            stepActivity = (Activity)repoStepActivity.CreateInstance();
+                            toAddStepActivity = true;
+                        }
+                        stepActivity.IsSharedRepositoryInstance = true;
+                    }
+                    else//Step not exist in Ginger repository so create new one
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [There's no need to call 'ToString()' on a string.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)